### PR TITLE
ReflectionClass modifications

### PR DESCRIPTION
**Pull request by asgrim**
Forms part of #134 and adds the following reflection modification methods:

 * setFinal
 * addMethod
 * removeMethod
 * addProperty
 * removeProperty

### CODE REVIEW

File: src/Reflection/ReflectionClass.php [hunk 1]
```diff
@@ -1127,4 +1127,111 @@ public function getAst()
...
+     *
+     * @param string $propertyName
+     * @return bool
+     */
+    public function removeProperty($propertyName)
+    {
+        $lowerName = strtolower($propertyName);
+        foreach ($this->node->stmts as $key => $stmt) {
+            if ($stmt instanceof PropertyNode) {
+                $propertyNames = array_map(function ($n) {
```
Comment (Ocramius) [inline src/Reflection/ReflectionClass.php]: Type hint on `$n`?

File: src/Reflection/ReflectionClass.php [hunk 2]
```diff
@@ -1127,4 +1127,111 @@ public function getAst()
...
+                    return strtolower($n->name);
+                }, $stmt->props);
+                if (in_array($lowerName, $propertyNames, true)) {
+                    $this->cachedProperties = null;
+                    unset($this->node->stmts[$key]);
+                    return true;
+                }
+            }
+        }
+        return false;
```
Comment (Ocramius) [inline src/Reflection/ReflectionClass.php]: 'bit of spacing around logical groups please

File: src/Reflection/ReflectionClass.php [hunk 3]
```diff
@@ -1127,4 +1127,111 @@ public function getAst()
...
+     *
+     * @param string $propertyName
+     * @param string $visibility
+     * @param bool $static
+     */
+    public function addProperty($propertyName, $visibility = 'public', $static = false)
+    {
+        $type = 0;
+        switch($visibility) {
+            case 'private':
```
Comment (Ocramius) [inline src/Reflection/ReflectionClass.php]: Constants to be used here, IMO. Could reuse internal reflection ones?

File: src/Reflection/ReflectionClass.php [hunk 4]
```diff
@@ -1127,4 +1127,111 @@ public function getAst()
...
+                $type |= ClassNode::MODIFIER_PRIVATE;
+                break;
+            case 'protected':
```
Comment (Ocramius) [inline src/Reflection/ReflectionClass.php]: same as above

File: src/Reflection/ReflectionClass.php [hunk 5]
```diff
@@ -1127,4 +1127,111 @@ public function getAst()
     {
         return $this->node;
     }
+
+    /**
+     * Set wether this class is final or not
+     *
+     * @param bool $shouldBeFinal
+     */
+    public function setFinal($shouldBeFinal)
```
Comment (Ocramius) [inline src/Reflection/ReflectionClass.php]: `should`?

**Comment by asgrim** [comment]
@Ocramius updated :)

**Comment by Ocramius** [comment]
:+1:

**Pull request merged**